packages/react-aria-components/src/Tree.tsx

```diff
@@ -469,6 +533,13 @@ function flattenTree<T>(collection: TreeCollection<T>, opts: TreeGridCollectionO
         if (hasContentNode) {
           clone.index = node?.index != null ? node?.index - 1 : 0;
         }
+
+        // For loader nodes that have a parent (aka non-root level loaders), these need their levels incremented by 1 for parity with their sibiling rows
+        // (Collection only increments the level if it is a "item" type node).
```

Comment:
Couldn't you render the loader as a child of the loading row instead of a sibling? That would give you more control over where it goes.

Reply:
It actually already is rendered as the child of the loading row (it has the proper parent, nextKey, prevKey etc), it is just that this line only adds an extra +1 for item type nodes: https://github.com/adobe/react-spectrum/blob/10bac541a2a5c666b443fe70f313455a405a5c77/packages/react-aria-components/src/Collection.tsx#L302

I could also have it do so for "loader" type elements, assuming that we then accept "loaders" as a common type node for collections in general (which now that I think about it again makes sense, was a bit wary of adding that to the base Collection implementation previously)

Reply:
resolved?